src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -265,51 +296,14 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
     setIsCollectionNftsLoading(isLoadingNext)
   }, [isLoadingNext, setIsCollectionNftsLoading])
 
-  const hasRarity = getRarityStatus(rarityStatusCache, collectionStats?.address, collectionNfts)
+  const hasRarity = useMemo(() => {
+    const hasRarity = getRarityStatus(rarityStatusCache, collectionStats?.address, collectionNfts) ?? false
+    setHasRarity(hasRarity)
+    return hasRarity
```

Comment:
what's the point of returning rarity here and then also setting rarity?  Wouldn't it make sense to just get the rarity from filters collection only or not set it at all

Reply:
this is where the value is retrieved from the queried collection and set in the hook